gax/src/test/java/com/google/api/gax/rpc/ClientContextTest.java

```diff
@@ -769,4 +786,76 @@ public void testExecutorSettings() throws Exception {
     transportChannel = (FakeTransportChannel) context.getTransportChannel();
     assertThat(transportChannel.getExecutor()).isSameInstanceAs(executorProvider.getExecutor());
   }
+
+  @Test
+  public void testAddApiKeyToHeadersFromStubSettings() throws IOException {
+    StubSettings settings = new FakeStubSettings.Builder().setApiKey("stub-setting-key").build();
+    EnvironmentProvider environmentProvider = new FakeEnvironmentProvider("env-key", null);
+    Map<String, String> headers = new HashMap<>();
+    ClientContext.addApiKeyToHeaders(settings, environmentProvider, headers);
+    assertTrue(headers.containsKey("x-goog-api-key"));
```

Comment:
Let's use Truth.

assertThat(headers.get("x-goog-api-key")).isEqualTo(...)

assertThat(headers).contains??? (Not sure if it has containsKey(). At least should be isTrue().)